internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -35,6 +40,172 @@ func (b *Builder) BuildLocalApp(workload *pbcatalog.Workload) *Builder {
 	return b
 }
 
+func buildTrafficPermissions(trustDomain string, workload *pbcatalog.Workload, computed *pbauth.ComputedTrafficPermissions) map[string]*pbproxystate.TrafficPermissions {
+	portsWithProtocol := workload.GetPortsByProtocol()
+
+	out := make(map[string]*pbproxystate.TrafficPermissions)
+	portToProtocol := make(map[string]pbcatalog.Protocol)
+	var allPorts []string
+	for protocol, ports := range portsWithProtocol {
+		for _, p := range ports {
+			allPorts = append(allPorts, p)
+			portToProtocol[p] = protocol
+			out[p] = &pbproxystate.TrafficPermissions{}
+		}
+	}
+
+	if computed == nil {
+		return out
+	}
```

Comment:
Does an empty traffic permission object result in an emtpy rbac filter which would mean deny all?

Reply:
Empty traffic permissions should defer to [`traffic_permission_default_allow`](https://github.com/hashicorp/consul/blob/802122640bb6a7686973d1d1480b55bc5f1a4816/proto-public/pbmesh/v1alpha1/proxy_state.proto#L50C8-L50C40). This should work based on a quick reading of https://github.com/hashicorp/consul/blob/main/agent/xdsv2/rbac_resources.go#L23

Reply:
Ok great.